dfir_lang/src/graph/ops/repeat_n.rs

```diff
@@ -0,0 +1,111 @@
+use quote::quote_spanned;
+
+use super::{
+    FloType, OperatorCategory, OperatorConstraints, OperatorWriteOutput, WriteContextArgs, RANGE_0,
+    RANGE_1,
+};
+
+/// Given a _bounded_ input stream, emits all values repeatedly over `N` iterations, in the same order.
+///
+/// Will cause `N` loop iterations.
+pub const REPEAT_N: OperatorConstraints = OperatorConstraints {
+    name: "repeat_n",
+    categories: &[OperatorCategory::Windowing],
+    hard_range_inn: RANGE_1,
+    soft_range_inn: RANGE_1,
+    hard_range_out: RANGE_1,
+    soft_range_out: RANGE_1,
+    num_args: 1,
+    persistence_args: RANGE_0,
+    type_args: RANGE_0,
+    is_external_input: false,
+    has_singleton_output: true,
+    flo_type: Some(FloType::Windowing),
+    ports_inn: None,
+    ports_out: None,
+    input_delaytype_fn: |_| None,
+    write_fn: |wc @ &WriteContextArgs {
+                   root,
+                   context,
+                   hydroflow,
+                   op_span,
+                   arguments,
+                   ident,
+                   is_pull,
+                   inputs,
+                   outputs,
+                   singleton_output_ident,
+                   ..
+               },
+               _diagnostics| {
+        let count_ident = wc.make_ident("count");
+
+        let write_prologue = quote_spanned! {op_span=>
+            #[allow(clippy::redundant_closure_call)]
+            let #singleton_output_ident = #hydroflow.add_state(
+                ::std::cell::RefCell::new(::std::vec::Vec::new())
+            );
+
+            // TODO(mingwei): Is this needed?
+            // Reset the value to the initializer fn if it is a new tick.
+            #hydroflow.set_state_tick_hook(#singleton_output_ident, move |rcell| { rcell.take(); });
+
+            let #count_ident = #hydroflow.add_state(::std::cell::Cell::new(0_usize));
+            #hydroflow.set_state_tick_hook(#count_ident, move |cell| { cell.take(); });
+        };
+
+        let vec_ident = wc.make_ident("vec");
+
+        let write_iterator = if is_pull {
+            // Pull.
+            let input = &inputs[0];
+            quote_spanned! {op_span=>
+                let mut #vec_ident = #context.state_ref(#singleton_output_ident).borrow_mut();
+                if #context.is_first_run_this_tick() {
+                    *#vec_ident = #input.collect::<::std::vec::Vec<_>>();
+                }
+                let #ident = std::iter::IntoIterator::into_iter(::std::clone::Clone::clone(&*#vec_ident));
+            }
+        } else if let Some(_output) = outputs.first() {
+            // Push with output.
```

Comment:
Why allow push with no output? Also is this checked at compile time or runtime? If runtime, can we fix that?